jina/types/document/graph.py

```diff
@@ -357,14 +357,38 @@ def to_dgl(self):
         """
         Construct a  `dgl.DGLGraph` from a `GraphDocument` instance.
         """
-        import dgl
+        from ... import JINA_GLOBAL
+
+        if JINA_GLOBAL.dgl_installed is None:
+            JINA_GLOBAL.dgl_installed = False
+            with ImportExtensions(
+                required=False,
+                pkg_name="dgl",
+                help_text=f"to_dgl method needs dgl",
+            ):
+                import dgl
+
+                JINA_GLOBAL.dgl_installed = True
+
+        if JINA_GLOBAL.torch_installed is None:
+            JINA_GLOBAL.torch_installed = False
+            with ImportExtensions(
+                required=False,
+                pkg_name="torch",
+                help_text=f"to_dgl method needs torch",
+            ):
+                import torch
+
+                JINA_GLOBAL.torch_installed = True
+
         import torch
+        import dgl
 
         source_nodes = torch.tensor(self.adjacency.row.copy())
         destination_nodes = torch.tensor(self.adjacency.col.copy())
 
         return dgl.graph((source_nodes, destination_nodes))
```

Comment:
I think we should add edge features frlm the graph proto and node dictionary as features to the nodes

Reply:
no, lets keep it like this and in a next iteration we can see how it should work